src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -783,6 +783,20 @@ public function prepare_item_for_response( $item, $request ) {
 					}
 
 					$size_data['source_url'] = $image_src[0];
+
+					if ( empty( $data['media_details']['sizes'][ $size ]['sources'] ) || ! is_array( $data['media_details']['sizes'][ $size ]['sources'] ) ) {
+						continue;
+					}
+
+					$size_data_sources = array();
+					$directory = dirname( $data['media_details']['sizes'][ $size ]['source_url'] );
+					foreach ( $data['media_details']['sizes'][ $size ]['sources'] as $mime => $mime_details ) {
+						$source_url                 = "{$directory}/{$mime_details['file']}";
+						$mime_details['source_url'] = $source_url;
+						$size_data_sources[ $mime ] = $mime_details;
+					}
+
+					$data['media_details']['sizes'][ $size ]['sources'] = $size_data_sources;
 				}
 
 				$full_src = wp_get_attachment_image_src( $post->ID, 'full' );
```

Comment:
Below here, you still need to ensure the `full` size also includes the `sources` associative array similar to the other size above. Also see my similar feedback on this being missing from the plugin in https://github.com/WordPress/performance/pull/443#discussion_r929943934.

We will need to get the information from `$data['media_details']['sources']` and use that to populate the `sources` for the `full` size here. Potentially we should even unset `$data['media_details']['sources']` then, since there would be no point to have this as duplicate information.

Reply:
Thank you for the review and details of the changes.

The PR now includes `sources` in the `full` size image and removes `$data['media_details']['sources']` from the array.